File: webapp/src/app/shared/model/game.ts
export interface Game {
  id: string;
  dateTime: string;
  playerOneId: string;
  playerOneName: string;
  playerOneRating: number;
  playerOneDeviation: number;
  playerTwoId: string;
  playerTwoName: string;
  playerTwoRating: number;
  playerTwoDeviation: number;
}

export interface CompletedGame extends Game {
  result: Result;
}

export interface ScheduledGame extends Game { }

export interface Result {
  playerOneScore: number;
  playerOneDeviationDelta: number;
  playerOneRatingDelta: number;
  playerTwoScore: number;
  playerTwoDeviationDelta: number;
  playerTwoRatingDelta: number;
}

File: api-service/src/main/kotlin/com/tm/rankme/query/game/Game.kt
package com.tm.rankme.query.game

import java.time.LocalDateTime

interface Game {
    val id: String
    val dateTime: LocalDateTime
    val playerOneId: String
    val playerOneName: String
    val playerOneRating: Int
    val playerOneDeviation: Int
    val playerTwoId: String
    val playerTwoName: String
    val playerTwoRating: Int
    val playerTwoDeviation: Int
}

class CompletedGame(
    override val id: String, override val dateTime: LocalDateTime,
    override val playerOneId: String, override val playerOneName: String,
    override val playerOneRating: Int, override val playerOneDeviation: Int,
    override val playerTwoId: String, override val playerTwoName: String,
    override val playerTwoRating: Int, override val playerTwoDeviation: Int,
    val result: Result
) : Game

class ScheduledGame(
    override val id: String, override val dateTime: LocalDateTime,
    override val playerOneId: String, override val playerOneName: String,
    override val playerOneRating: Int, override val playerOneDeviation: Int,
    override val playerTwoId: String, override val playerTwoName: String,
    override val playerTwoRating: Int, override val playerTwoDeviation: Int
) : Game

data class Result(
    val playerOneScore: Int,
    val playerOneDeviationDelta: Int,
    val playerOneRatingDelta: Int,
    val playerTwoScore: Int,
    val playerTwoDeviationDelta: Int,
    val playerTwoRatingDelta: Int
)